src/auction/index.js

```diff
@@ -0,0 +1,54 @@
+import db, { AUCTION_STORE, IG_STORE } from '../utils/db.js';
+import validate from '../utils/validation.js';
+import types from './types.js';
+import {
+	getBids,
+	getEligibleBuyers,
+	getScores,
+	uuidv4,
+} from './utils.js';
+
+/*
+ * @function
+ * @name runAdAuction
+ * @description run an auction
+ * @author Newton <cnewton@magnite.com>
+ * @param {object} options - An object of options to configure an auction
+ * @throws {Error} Any parameters passed are incorrect or an incorrect type
+ * @return {null | Promise<Token>}
+ *
+ * @example
+ *   runAdAuction({ seller: 'foo', decision_logic_url: 'http://example.com/auction', interst_group_buyers: [ 'www.buyer.com' ] });
+ */
+export default async function runAdAuction (conf) {
+	validate.param(conf, 'object');
+	validate.hasRequiredKeys(conf, [ 'seller', 'decision_logic_url', 'interest_group_buyers' ]);
+	validate.hasInvalidOptionTypes(conf, types);
+
+	// console.info('get all interest groups');
+	const buyers = await db.store.getAll(IG_STORE);
+
+	// console.info('checking eligibility of buyers based on "interest_group_buyers"');
+	const eligibleBuyers = getEligibleBuyers(buyers, conf.interest_group_buyers);
```

Comment:
```suggestion
	const eligibleIntestGroups = getEligibleInterestGroups(interestGroups, conf.interest_group_buyers);
```
or maybe `filterEligibleBuyers`